xapian-applications/omega/postprocess

```diff
@@ -0,0 +1,246 @@
+
+        test_search.write('821f03288846297c2cf43c34766a38f7,"book","45,54",0\n')
+        test_search.write('821f03288846297c2cf43c34766a38f7,"book","45,54",0\n')
+        test_search.write('d41d8cd98f00b204e9800998ecf8427e,"","",0\n')
+        test_search.write('098f6bcd4621d373cade4e832627b4f6,"test","35,47",0\n')
+
+        test_search.flush()
+
+        test_clicks = tempfile.NamedTemporaryFile(delete=True)
+
+        test_clicks.write('821f03288846297c2cf43c34766a38f7,54\n')
+        test_clicks.write('821f03288846297c2cf43c34766a38f7,54\n')
+        test_clicks.write('098f6bcd4621d373cade4e832627b4f6,35\n')
+        test_clicks.flush()
+
+        test_final = tempfile.NamedTemporaryFile(delete=True)
+
+        generate_combined_log(test_search.name, test_clicks.name, test_final.name)
+
+        test_final.flush()
+
+        # Test entries in final.log are correct.
+        with open(test_final.name, 'r') as final_f:
+            reader = csv.reader(final_f)
+
+            # Skip header row.
+            next(final_f)
+
+            for i, row in enumerate(reader):
+                if i == 0:
+                    assert row == ['821f03288846297c2cf43c34766a38f7',
+                                  'book', '45,54', '0', '45:0,54:2'], "Incorrect entry in final.log"
+                if i == 1:
+                    assert row == ['098f6bcd4621d373cade4e832627b4f6',
+                                  'test', '35,47', '0', '35:1,47:0'], "Incorrect entry in final.log"
+
+        test_query = tempfile.NamedTemporaryFile(delete=True)
+
+        generate_query_file(test_final.name, test_query.name)
+
+        test_query.flush()
+
+        # Test entries in query.txt are correct.
+        with open(test_query.name, 'r') as query_f:
+            reader = csv.reader(query_f)
+
+            for i, row in enumerate(reader):
+                if i == 0:
+                    assert row == ['821f03288846297c2cf43c34766a38f7','book'], "Incorrect entry in query.txt"
+                if i == 1:
+                    assert row == ['098f6bcd4621d373cade4e832627b4f6','test'], "Incorrect entry in query.txt"
+    finally:
+        # Close all files to safely delete them.
+        test_search.close()
+        test_clicks.close()
+        test_final.close()
+        test_query.close()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='''Postprocess click data files. This script generates the
+        final clickstream log file from input search and click log files and
+        creates query file that can be used by the Xapian Letor module for
+        generating its training files.''')
+    parser.add_argument("search_log", type=str, help="Path to the search.log file.")
+    parser.add_argument("clicks_log", type=str, help="Path to the clicks.log file.")
+    parser.add_argument("final_log", type=str, help="Path to save final.log file.")
+    parser.add_argument("query_file", type=str, help="Path to save query.txt file.")
```

Comment:
I think for these you should talk about "the search log file" and so forth, rather than "the search.log file". I think it'll be clearer, because it doesn't second guess what people call the files.